admin-dev/themes/new-theme/js/pages/product/components/categories/tags.ts

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import ProductMap from '@pages/product/product-map';
+import ProductEventMap from '@pages/product/product-event-map';
+import {EventEmitter} from "events";
+
+const ProductCategoryMap = ProductMap.categories;
+
+export default class Tags {
+  eventEmitter: EventEmitter;
+
+  container: HTMLElement;
+
+  constructor(
+    eventEmitter: EventEmitter,
+    containerSelector: string,
+  ) {
+    this.eventEmitter = eventEmitter;
+    const container = document.querySelector(containerSelector);
+
+    if (!(container instanceof HTMLElement)) {
+      throw '"container" must be a valid HTMLElement';
+    }
+
+    this.container = container;
+  }
+
+  public render(categories: Array<Category>): void {
```

Comment:
```suggestion
  public render(categories:Category[]): void {
```

@NeOMakinG I like this notation better, but it's put to you? You think `Array<>` is clearer?